src/Security/Voter/UserRequestVoter.php

```diff
@@ -17,30 +16,41 @@ public function __construct(private readonly AuthorizationCheckerInterface $chec
 
     protected function supports(string $attribute, mixed $subject): bool
     {
-        return in_array($attribute, [Permissions::UPDATE, Permissions::READ]) && $subject instanceof UserRequest;
+        return in_array($attribute, [Permissions::UPDATE, Permissions::READ, Permissions::DELETE]) && $subject instanceof UserRequest;
     }
 
     protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token): bool
     {
+        /** @var User $user */
         $user = $token->getUser();
-        if (!$user instanceof UserInterface || !$subject instanceof UserRequest) {
+        if (!$user instanceof User || !$subject instanceof UserRequest) {
```

Comment:
Just so that you know, here it seems OK, but this isntance of User could cause problems 
For example, if you use stateless API tokens, or client_credentials, you may have something like an instance of NullUser that implements UserInterface but that is not User. For now, this is OK here